### PR TITLE
[diffusion] Support MiniMax-H3 FastH3 Preview v1 LoRA (Dense + VSA-H3)

### PR DESCRIPTION
**Pull request by niehen6174**
## Motivation

FastH3 Preview v1 is a 4-forward distilled student of MiniMax-H3 ([blog](https://haoailab.com/blogs/fasth3-preview/), [adapter](https://huggingface.co/FastVideo/FastVideo-FastH3-4-step-Preview-v1-LoRA)).

The file is not generic PEFT: rank-64 A/B plus whole-parameter `.diff` / `.diff_b` / `.set_weight` (VSA also ships trained `to_gate_compress`).
Official VSA also cannot run on dense FA or Wan VSA (tile 256); it needs packed `[text|cond|audio|video]` attention, tile 64, ~90% sparsity, and prefix-exempt tiles.

This PR teaches the LoRA loader the extra key kinds and adds the official VSA-H3 backend so both FastH3 variants can run in-tree.

## Changes

- **LoRA format.** `normalize_lora_state_dict` drops whole-parameter keys. `LoRAPipeline` extracts them via `AdapterDensePayload`, merges A/B as usual, then applies `.diff` (add) / `.set_weight` (replace). Merge cache is disabled only when a payload is present (cache is base+A/B only). Ordinary A/B LoRAs are unchanged. FastVideo SwiGLU `fc1.lora_B` `[value; gate]` → native `[gate; value]` stays the existing name-gated swap.
- **VSA-H3 backend.** New `video_sparse_attn_h3` (`vsa_h3` / `vsa-h3` aliases). Attention kernel is Triton `fastvideo_kernel.block_sparse_attn_from_indices` (tile 64). The import from Wan `video_sparse_attn` is tile/pad index helpers only, not `vsa.video_sparse_attn`. Token refiner stays FA. Optional `to_gate_compress`; missing base weights stay zero.
- **Denoise / DiT.** Reconstruct packed prefix from `text_pos_info` / `update_mask` / `audio_pos_info`. Ulysses all-to-alls `gate_compress` with Q (must divide 56 heads). Ring stays FA-only.
- **Mapping.** `to_gate_compress` in H3 `param_names_mapping`.

## E2E (H200, T2VA 768p 16:9, 5s, seed 1000, official video=12 / audio=3)

Same prompt / seed / resolution / scheduler.
Times are the **second generate** after one discarded warmup (FastH3 Dense / official: FA; FastH3 VSA: VSA-H3).
E2E is encode + denoise + decode + save; load is not included.
prompt: `"integrated_multimodal_description: A red fox runs through fresh snow at dawn. overall_soundscape: Fast pawsteps in snow, winter wind, and distant birds."` 

| Case | Steps | Denoise | s/step | E2E | Peak VRAM |
|---|---|---|---|---|---|
| FastH3 Dense | 4 | 16.3s | 4.1s | 29.2s | 78.1 GB |
| FastH3 VSA-H3 | 4 | 14.2s | 3.6s | 23.1s | 86.9 GB |
| Official H3 FA | 20 | 97.9s | 5.2s | 110.3s | 78.7 GB |

**FastH3 Dense**

https://github.com/user-attachments/assets/eecd65c5-9bf7-4cbb-8939-0fbb7402909c

**FastH3 VSA-H3**

https://github.com/user-attachments/assets/4d48936c-40da-468a-aaaf-29458b8d3dbf

**Official H3 FA**

https://github.com/user-attachments/assets/bf96722c-2c67-40ae-9cbc-f5fe1437e2fc

## Examples

Adapter: [FastVideo/FastVideo-FastH3-4-step-Preview-v1-LoRA](https://huggingface.co/FastVideo/FastVideo-FastH3-4-step-Preview-v1-LoRA).
T2VA only. Use `--num-inference-steps 5` (5 sigma points / 4 DiT forwards).
Do **not** pass `--flow-shift`; official FastH3 keeps H3's video=12 / audio=3 scheduler.
`--lora-alpha 64 --lora-scale 1.0`. VSA needs `fastvideo-kernel` (Triton 64-token path).

**Dense / Data-Free** (FA):

```bash
sglang generate \
  --model-path MiniMaxAI/MiniMax-H3 \
  --model-variant fl2va \
  --lora-path FastVideo/FastVideo-FastH3-4-step-Preview-v1-LoRA \
  --lora-weight-name dense-datafree/adapter_model.safetensors \
  --lora-alpha 64 --lora-scale 1.0 \
  --attention-backend fa \
  --num-inference-steps 5 --seed 1000 \
  --prompt "integrated_multimodal_description: A red fox runs through fresh snow at dawn. overall_soundscape: Fast pawsteps in snow, winter wind, and distant birds." \
  --save-output
```

**VSA / Data-Free** (tile 64, ~90% sparsity). Point the transformer at `video_sparse_attn_h3` (`vsa_h3` / `vsa-h3` also work). Token refiner stays FA.

```bash
pip install 'fastvideo-kernel==0.3.5' --no-deps   # Triton path; skip sm90 ThunderKittens

sglang generate \
  --model-path MiniMaxAI/MiniMax-H3 \
  --model-variant fl2va \
  --lora-path FastVideo/FastVideo-FastH3-4-step-Preview-v1-LoRA \
  --lora-weight-name vsa-datafree/adapter_model.safetensors \
  --lora-alpha 64 --lora-scale 1.0 \
  --attention-backend video_sparse_attn_h3 \
  --attention-backend-config VSA_sparsity=0.9,tile_size=64 \
  --component-attention-backends transformer=video_sparse_attn_h3 \
  --num-inference-steps 5 --seed 1000 \
  --prompt "integrated_multimodal_description: A red fox runs through fresh snow at dawn. overall_soundscape: Fast pawsteps in snow, winter wind, and distant birds." \
  --save-output
```

Do not run the VSA adapter on dense FA. Official FastH3 forbids it.

**Official H3** (no LoRA, FA, 20 steps) used as the timing / quality baseline above:

```bash
sglang generate \
  --model-path MiniMaxAI/MiniMax-H3 \
  --model-variant fl2va \
  --attention-backend fa \
  --num-inference-steps 20 --seed 1000 \
  --prompt "integrated_multimodal_description: A red fox runs through fresh snow at dawn. overall_soundscape: Fast pawsteps in snow, winter wind, and distant birds." \
  --save-output
```

## Test plan

- [x] `pytest` `test_fasth3.py` + H3 contract/admission
- [x] Dense official T2VA 768p 5pt (warmup + timed)
- [x] VSA official T2VA 768p 5pt (warmup + timed)
- [x] Official H3 FA T2VA 768p 20pt (FA warmup + timed)

---
### CI States

Latest PR Test (Base): :x: [Run #33609163720](https://github.com/sgl-project/sglang/actions/runs/33609163720)
Latest PR Test (Extra): :x: [Run #33609163436](https://github.com/sgl-project/sglang/actions/runs/33609163436)
Latest PR Test (AMD ROCm 7.2): :x: [Run #33609163726](https://github.com/sgl-project/sglang/actions/runs/33609163726)

### CODE REVIEW

**Comment by niehen6174** [comment]
#35807 supports fastH3, so this PR can be closed.
